src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -49,25 +50,44 @@ private void refreshCacheItem(final MessageCacheItem cacheItem, Iterator<DataSou
 			return;
 		}
 
-		long timestampOld = cacheItem.getTimestamp();
 		DataSourceEnum dataSource = msgSourceQueueIter.next();
-		String localeOrig = dto.getLocale();
-		if (dataSource.equals(DataSourceEnum.VIP) && dto.getLocale().equals(ConstantsKeys.SOURCE)) {
-			dto.setLocale(ConstantsKeys.LATEST);
-		}
-		dataSource.createMessageOpt(dto).getComponentMessages(cacheItem);
-		long timestamp = cacheItem.getTimestamp();
-		if (timestampOld == timestamp) {
-			logger.debug(FormatUtils.format(ConstantsMsg.GET_MESSAGES_FAILED, dto.getComponent(), dto.getLocale(), dataSource.toString()));
-		}
-		dto.setLocale(localeOrig);
+		if (!proceed(dataSource)) {
+			refreshCacheItem(cacheItem, msgSourceQueueIter); // Try the next dataSource
+		} else {
```

Comment:
else is unnecessary because there is no code after else block. Removing it makes easy to read.